WordPress/gutenberg
# [Webfonts] Rename to Fonts API

Closes #42868.

## What?

Renames the API from Webfonts to Fonts API.

## Why?


The renaming is to align the feature to well-known authoritative standards for consistency and maintainability. Why? These standards do not use "webfont" to describe `@font-face` CSS styles. Rather, they use terms such as "font", "custom font", "downloadable font", and "web font".

## How?

1. All publicly facing functionality is deprecated as part of the temporary BC layer. Notices will be thrown to alert developers where to upgrade their code to the new API.
2. "web" is removed from all inline comments, docblocks, and code names (except for the original which are now deprecated).

## Testing Instructions





### Testing Instructions for Keyboard


## Testing Instructions

1. Active TT3
2. Activate the Gutenberg plugin
3. Reset the global styles to the default by:
   - Go to the Site Editor > Global Styles "Styles Beta" UI
   - Click on the 3 vertical dots
   - Select "Reset to defaults"
   - Click on the "Save" button
4. Examine the dynamically generated `@font-face` stylings in the admin area by:
   - Open Dev Tools in your favorite browser
   - Search for `wp-fonts-local` (which is in the `<head>`)
   - Expand the `<style>` element to inspect the CSS
   - The CSS should be:
```html
<style id="wp-fonts-local">
@font-face{font-family:"Source Serif Pro";font-style:normal;font-weight:200 900;font-display:fallback;src:local("Source Serif Pro"), url('/wp-content/themes/twentytwentytwo/assets/fonts/source-serif-pro/SourceSerif4Variable-Roman.ttf.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"Source Serif Pro";font-style:italic;font-weight:200 900;font-display:fallback;src:local("Source Serif Pro"), url('/wp-content/themes/twentytwentytwo/assets/fonts/source-serif-pro/SourceSerif4Variable-Italic.ttf.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"DM Sans";font-style:normal;font-weight:400;font-display:fallback;src:local("DM Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/dm-sans/DMSans-Regular.ttf') format('truetype');}
@font-face{font-family:"DM Sans";font-style:italic;font-weight:400;font-display:fallback;src:local("DM Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/dm-sans/DMSans-Italic.ttf') format('truetype');}
@font-face{font-family:"DM Sans";font-style:normal;font-weight:700;font-display:fallback;src:local("DM Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/dm-sans/DMSans-Bold.ttf') format('truetype');}
@font-face{font-family:"DM Sans";font-style:italic;font-weight:700;font-display:fallback;src:local("DM Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/dm-sans/DMSans-BoldItalic.ttf') format('truetype');}
@font-face{font-family:"IBM Plex Sans";font-style:normal;font-weight:400;font-display:fallback;src:local("IBM Plex Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexSans-Light.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Sans";font-style:italic;font-weight:400;font-display:fallback;src:local("IBM Plex Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexSans-LightItalic.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Sans";font-style:normal;font-weight:200;font-display:fallback;src:local("IBM Plex Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexSans-ExtraLight.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Sans";font-style:italic;font-weight:200;font-display:fallback;src:local("IBM Plex Sans"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexSans-ExtraLightItalic.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Mono";font-style:normal;font-weight:400;font-display:fallback;src:local("IBM Plex Mono"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexMono-Text.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Mono";font-style:italic;font-weight:400;font-display:fallback;src:local("IBM Plex Mono"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexMono-TextItalic.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Mono";font-style:normal;font-weight:700;font-display:fallback;src:local("IBM Plex Mono"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexMono-Bold.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:"IBM Plex Mono";font-style:italic;font-weight:700;font-display:fallback;src:local("IBM Plex Mono"), url('/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/IBMPlexSans-BoldItalic.woff2') format('woff2');font-stretch:normal;}
@font-face{font-family:Inter;font-style:normal;font-weight:200 900;font-display:fallback;src:local(Inter), url('/wp-content/themes/twentytwentytwo/assets/fonts/inter/Inter.ttf') format('truetype');font-stretch:normal;}
</style>
```
4. Examine the front-end by:
   - Go to the front-end of the website
   - Open Dev Tools in your favorite browser
   - Search for `wp-fonts-local` 
   - Expand the `<style>` element to inspect the CSS
   - The CSS should be the same as above.
5. Change to the Blue style variation and modify the "text" typography by:
    - Go back to the Global Styles UI in the admin
    - Select "Browser styles"
    - Select the "blue" variation
    - Select "Browse styles" to go back
    - Select "Typography"
    - Select "Text"
    - Change the size to "4"
    - Change the "Appearance" to "Semi Bold Italic"
    - Inspect the CSS as you did in Step 3. It should be the same.
6. Examine the front-end by:
    - Refresh (may have to refresh multiple times until the blue styling renders)
    - Search for `wp-fonts-local`
   - Expand the `<style>` element to inspect the CSS and verify the CSS.

### Testing a plugin

[This gist is a plugin](https://gist.github.com/hellofromtonya/1bbbc9a5354a905278080a9f022c2f60) that uses Jetpack's Google Font Provider package, which is using the original (deprecated) API functionality and web fonts array structure. So it should continue to work and throw deprecation notices.

Instructions of how to install and test are in the plugin's `README.md` file, which [you can get here](https://gist.github.com/hellofromtonya/1bbbc9a5354a905278080a9f022c2f60).

Note: This plugin has a Google Fonts Provider which means it will print `<style id="wp-fonts-jetpack-google-fonts">..</style>` into the `<head>`.
